R/todor.R

```diff
@@ -102,9 +105,37 @@ todor <- function(todo_types = NULL, search_path = getwd(), file = NULL) {
   processed <- lapply(files, function(x) process_file(x, todo_types))
   names(processed) <- files
   markers <- create_markers(processed)
-  build_rstudio_markers(markers)
+
+  if(!(output %in% c("markers","text"))){
+    stop("Output format not recognised. Available options are \"markers\" and \"text\"")
+  }
+
+  if (output == "text") {
+    return(build_report(markers))
+  } else {
+    build_rstudio_markers(markers)
+  }
 }
 
+build_report <- function(markers){
+
+  files <- unique(unlist(lapply(markers, "[", "file")))
+
+  get <- function(file){
+    paste0("**", R.utils::getRelativePath(file), "** \n\n",
+           paste0("- ", sapply(Filter(function(x) x$file==file, markers), `[[`, "message"), collapse = "\n"),
+           "\n\n"
+    )
+  }
+
+  text <- paste(unlist(purrr::map(files, get)), collapse = "")
```

Comment:
in general, I don't mind purrr here, but looks like base `Map` will do the same, so loading whole package just for this functionality looks like overhead, could you call `Map(get, files)` instead (with new `get` name)?

Reply:
Yeah, that makes a lot of sense! `purrr` is just my go-to - will swap it out for `Map`

Reply:
`purrr::map()` now replaced with `Map()`